src/dbally/collection/collection.py

```diff
@@ -143,12 +182,154 @@ def list(self) -> Dict[str, str]:
+
+        Args:
+            selected_view_name: The name of the selected view.
+            question: The question to be answered.
+            event_tracker: The event tracker for logging and tracking events.
+            llm_options: Options for the LLM client.
+            dry_run: If True, only generate the query without executing it.
+
+        Returns:
+            Any: The result from the selected view.
+        """
+        selected_view = self.get(selected_view_name)
+        view_result = await selected_view.ask(
+            query=question,
+            llm=self._llm,
+            event_tracker=event_tracker,
+            n_retries=self.n_retries,
+            dry_run=dry_run,
+            llm_options=llm_options,
+        )
+        return view_result
+
+    async def _generate_textual_response(
+        self,
+        view_result: ViewExecutionResult,
+        question: str,
+        event_tracker: EventTracker,
+        llm_options: Optional[LLMOptions],
+    ) -> str:
+        """
+        Generate a textual response from the view result.
+
+        Args:
+            view_result: The result from the view.
+            question: The question to be answered.
+            event_tracker: The event tracker for logging and tracking events.
+            llm_options: Options for the LLM client.
+
+        Returns:
+            The generated textual response.
+        """
+        textual_response = await self._nl_responder.generate_response(
+            result=view_result,
+            question=question,
+            event_tracker=event_tracker,
+            llm_options=llm_options,
+        )
+        return textual_response
+
+    async def _handle_fallback(
+        self,
+        question: str,
+        dry_run: bool,
+        return_natural_response: bool,
+        llm_options: Optional[LLMOptions],
+        selected_view_name: str,
+        event_tracker: EventTracker,
+        caught_exception: HANDLED_EXCEPTION_TYPES,
+    ) -> ExecutionResult:
+        """
+        Handle fallback if the main query fails.
+
+        Args:
+            question: The question to be answered.
+            dry_run: If True, only generate the query without executing it.
+            return_natural_response: If True, return the natural language response.
+            llm_options: Options for the LLM client.
+            selected_view_name: The name of the selected view.
+            event_tracker: The event tracker for logging and tracking events.
+            caught_exception: The exception that was caught.
+
+        Returns:
+            The result from the fallback collection.
+
+        """
+
+        fallback_event = FallbackEvent(
+            triggering_collection_name=self.name,
+            triggering_view_name=selected_view_name,
+            fallback_collection_name=self._fallback_collection.name,
+            error_description=repr(caught_exception),
+        )
+
+        async with event_tracker.track_event(fallback_event) as span:
```

Comment:
let's test how it looks in OTEL / Langsmith when we nest events inside, not sure if it is supported right now